napari/layers/shapes/_tests/test_shapes.py

```diff
@@ -476,6 +505,71 @@ def test_ellipses():
     assert layer.ndim == shape[2]
     assert np.all([s == 'ellipse' for s in layer.shape_type])
 
+    # Test single four corner (vertices, shape_type) tuple
+    shape = (1, 4, 2)
+    np.random.seed(0)
+    vertices = 20 * np.random.random(shape)
+    data = (vertices, "ellipse")
+    layer = Shapes(data)
+    assert layer.nshapes == shape[0]
+    assert np.all(layer.data[0] == data[0])
+    assert layer.ndim == shape[2]
+    assert np.all([s == 'ellipse' for s in layer.shape_type])
+
+    # Test multiple four corner (list of vertices, shape_type) tuple
+    shape = (10, 4, 2)
+    np.random.seed(0)
+    vertices = 20 * np.random.random(shape)
+    data = (vertices, "ellipse")
+    layer = Shapes(data)
+    assert layer.nshapes == shape[0]
+    assert np.all([np.all(ld == d) for ld, d in zip(layer.data, vertices)])
+    assert layer.ndim == shape[2]
+    assert np.all([s == 'ellipse' for s in layer.shape_type])
+
+    # Test list of four corner (vertices, shape_type) tuples
+    shape = (10, 4, 2)
+    np.random.seed(0)
+    vertices = 20 * np.random.random(shape)
+    data = [(vertices[i], "ellipse") for i in range(shape[0])]
+    layer = Shapes(data)
+    assert layer.nshapes == shape[0]
+    assert np.all([np.all(ld == d) for ld, d in zip(layer.data, vertices)])
+    assert layer.ndim == shape[2]
+    assert np.all([s == 'ellipse' for s in layer.shape_type])
```

Comment:
Can you make these separate tests instead of bunching them all together? That makes it much easier to pinpoint failures when they happen in the future.